pkg/apis/batch/types.go

```diff
@@ -244,6 +244,16 @@ type CronJobSpec struct {
 	// JobTemplate is the object that describes the job that will be created when
 	// executing a CronJob.
 	JobTemplate JobTemplateSpec
+
+	// The number of successful finished jobs to retain.
+	// This is a pointer to distinguish between explicit zero and not specified.
+	// +optional
+	SuccessfulJobsHistoryLimit *int32
+
+	// The number of failed finished jobs to retain.
+	// This is a pointer to distinguish between explicit zero and not specified.
+	// +optional
+	FailedJobsHistoryLimit *int32
```

Comment:
I think for CronJob, which can generate a lot of successful jobs, it makes more sense to use two separate limits for success/failure.

If we use a single limit, with CronJobs that have a frequent schedule and in the normal situation where failures are infrequent, failed jobs are likely to get cleaned up before someone has time to look at them as more successful jobs pile up every minute or 5min or so.